flow-sample-orientation-lock/src/main/java/flow/sample/orientation/OrientationSampleDispatcher.java

```diff
@@ -0,0 +1,132 @@
+import android.support.annotation.NonNull;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import flow.Dispatcher;
+import flow.Traversal;
+import flow.TraversalCallback;
+
+import static android.content.pm.ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE;
+import static android.content.pm.ActivityInfo.SCREEN_ORIENTATION_PORTRAIT;
+import static android.content.pm.ActivityInfo.SCREEN_ORIENTATION_REVERSE_LANDSCAPE;
+import static android.content.pm.ActivityInfo.SCREEN_ORIENTATION_REVERSE_PORTRAIT;
+import static android.content.pm.ActivityInfo.SCREEN_ORIENTATION_SENSOR_LANDSCAPE;
+import static android.content.pm.ActivityInfo.SCREEN_ORIENTATION_SENSOR_PORTRAIT;
+import static android.content.pm.ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED;
+
+final class OrientationSampleDispatcher implements Dispatcher {
+  /**
+   * Using a static for this is a fragile hack for demo purposes. In
+   * real life you'd want something hanging off of a custom application
+   * class--perhaps a Mortar activity scope.
+   */
+  private static TraversalCallback hangingCallback;
+
+  public static void finishPendingTraversal() {
+    if (hangingCallback != null) {
+      // The previous dispatcher was unable to finish its traversal because it
+      // required a configuration change. Let flow know that we're awake again
+      // in our new orientation and that traversal is done.
+
+      TraversalCallback ref = hangingCallback;
+      hangingCallback = null;
+      ref.onTraversalCompleted();
+    }
+  }
+
+  private final Activity activity;
+
+  OrientationSampleDispatcher(Activity activity) {
+    this.activity = activity;
+  }
+
+  @Override
+  public void dispatch(@NonNull Traversal traversal, @NonNull TraversalCallback callback) {
+    Log.d("BasicDispatcher", "dispatching " + traversal);
+    final Object leaving = traversal.origin == null ? null : traversal.origin.top();
+    final Object entering = traversal.destination.top();
+
+    ViewGroup frame = (ViewGroup) activity.findViewById(R.id.basic_activity_frame);
+
+    @LayoutRes final int layout;
+    if (entering instanceof LandscapeScreen) {
+      layout = R.layout.landscape_screen;
+    } else if (entering instanceof LooseScreen) {
+      layout = R.layout.loose_screen;
+    } else {
+      throw new AssertionError("Unrecognized screen " + entering);
+    }
+
+    boolean outgoingWasLocked = leaving instanceof LandscapeScreen;
+    boolean incomingNeedsLock = entering instanceof LandscapeScreen;
+
+    boolean waitForOrientationChange = incomingNeedsLock && requestLandscapeLock();
+
+    if (waitForOrientationChange) {
+      // There is about to be an orientation change, which means there will
+      // soon be a new activity and a new dispatcher. Let them complete
+      // this traversal, so that we don't try to show a landscape-only screen
+      // in portrait.
+      hangingCallback = callback;
```

Comment:
Do we maybe want a `callback.suspend()`? It would be terminal for that callback instance, just like `onCallbackCompleted`. Flow could handle the `finishPendingTraversal` part.